app/models/project.rb

```diff
@@ -96,27 +96,20 @@ def in_mediaflux?
     mediaflux_id.present?
   end
 
+  # This method narrows the list down returned by `all_projects` to only those projects where the user has
+  # been given a role (e.g. sponsor, manager, or data user.) For most users `all_projects` and `user_projects`
+  # are identical, but for administrators the lists can be very different since they are not part of most
+  # projects even though they have access to them in Mediaflux.
   def self.users_projects(user)
-    # See https://scalegrid.io/blog/using-jsonb-in-postgresql-how-to-effectively-store-index-json-data-in-postgresql/
-    # for information on the @> operator
-    uid = user.uid
-    query_ro = '{"data_user_read_only":["' + uid + '"]}'
-    query_rw = '{"data_user_read_write":["' + uid + '"]}'
-    query = "(metadata_json @> ? :: jsonb) OR (metadata_json @> ? :: jsonb) OR (metadata_json->>'data_sponsor' = ?) OR (metadata_json->>'data_manager' = ?)"
-    args = [query_ro, query_rw, uid, uid]
-    Project.where( query, *args)
-  end
-
-  def self.sponsored_projects(sponsor)
-    Project.where("metadata_json->>'data_sponsor' = ?", sponsor)
-  end
-
-  def self.managed_projects(manager)
-    Project.where("metadata_json->>'data_manager' = ?", manager)
+    all_projects(user).select do |project|
+      project[:data_manager] == user.uid || project[:data_sponsor] == user.uid || project[:data_users].include?(user.uid)
+    end
```

Comment:
Filter the projects that came from Mediaflux. 

Notice that Mediaflux filters projects that the user has access to on the backend, in here we just make sure the user has a roles on the project (manager, sponsor, data user.) This extra filtering is necessarily because system administrators have access to other projects in Mediaflux even when they don't have a role in them.